conanfile.py

```diff
@@ -0,0 +1,47 @@
+from conans import ConanFile, tools, CMake
+
+
+class mediagraphConan(ConanFile):
+    name = "mediagraph"
+    license = "Custom"
+    url = "https://github.com/jpilet/mediagraph"
+    description = "A multi-threaded media pipelining library"
+    author = "Julien Pilet <jpilet@opticode.ch>"
+    topics = ("conan", "pipeline", "media", "multi-threading")
+    
+    exports = ("LICENSE.md", "README.md")
+    exports_sources = ("CMakeLists.txt", "*.cpp", "*.h", "types/*")
+    
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False]}
```

Comment:
It's pretty cool that conan handles this automatically with CMake: https://bincrafters.readthedocs.io/en/latest/howtos/add_fpic_option_to_recipe.html

Reply:
is it propagating the fpic requirement in all the dependencies of any lib linked dynamically?
that would be nice!

Reply:
yes, for CMake-based project's it's setting CMAKE_POSITION_INDEPENDENT_CODE, which then acts on all targets

Reply:
hum let's assume I want to make a .so that links with a static version of opencv that itself depends on a bunch of libs, libz for example. Will conan automatically propagate fPIC to opencv and all its deps?

Reply:
that I don't know, I think not. Rather one has to make sure that the dependent libs are compiled with the option fPIC.

Reply:
that's very good already, if it can say that there's something wrong before trying to link the .so. cool.